# Null hypothesis significance testing

![](http://slideplayer.com/slide/3791615/13/images/1/Hypothesis+Testing+Steps+in+Hypothesis+Testing%3A.jpg)

http://www.rpubs.com/rfpopp/data_analysis_lab8




### Example 1
> Suppose a coin toss turns up 12 heads out of 20 trials. At .05 significance level, can one reject the null hypothesis that the coin toss is fair?

In [1]:
12/20

[1] 0.6

#### Steps:
1. State the null and alternative hypotheses:
        HO: p = 0.5
        H1: p != 0.5
2. Choose level of significance: 0.05
3. Find critical value

In [4]:
qt(c(0.25, 0.975), df=19)

[1] -0.6876215  2.0930241

4. calculate test statitic

In [13]:
#### Test Statistice ####

pbar = 12 / 20       # sample proportion
p = .5               # hypothesized value
n = 20               # sample size

# test statistic
z = (pbar - p)/sqrt(p*(1-p)/n)
z

[1] 0.8944272

5. Draw your conclusion 

Because 0.89 is not a the in the rejection region (defined by the critical values), we fail to reject the null hypothesis

##### Approach # 2

In [14]:
prop.test(12, 20, p=0.5, correct=FALSE)


	1-sample proportions test without continuity correction

data:  12 out of 20, null probability 0.5
X-squared = 0.8, df = 1, p-value = 0.3711
alternative hypothesis: true p is not equal to 0.5
95 percent confidence interval:
 0.3865815 0.7811935
sample estimates:
  p 
0.6 


Per the `X-squared` value (0.8), the value is not in the rejection region, __therfore we fail to reject H0.__ We can also say that __we cannot reject the null hypothesis at a 0.05 significance level.__ 

### Example 2
>David claims that the weather forecasts produced by the local radio are not better than those achieved by tossing a fair coin and predicting rain is a head is obtained or no rain if a tail is obtained. He records the weather for 30 randomly selected days. The local radio forecast is correct on 21 of these days. 

Calculate the proportion:

In [15]:
21/30

[1] 0.7

p = 0.7

#### Steps:
1. State the null and alternative hypothesis:

       HO: p = 0.5
       HA: p >= 0.5
2. Chose the level of siginificant 0.05
3. Find the critical value by finding the 2.5th and the 97.5th percentile of the Student t distribution with 29 df

In [22]:
qt(c(0.05/2, 1-(0.05/2)), df = (30-1))

[1] -2.04523  2.04523

The 2.5th and 97.5th percentiles of the Student t distribution with 29 degrees of freedom are -2.04452 and 2.0452 respectively.

4. calcuate test statistic

![](https://cdn-images-1.medium.com/max/800/1*K8QeUa6X28OXashLRNidRQ.png)


In [31]:
xbar <- 21/30    # sample proportion
p <- 0.5        # hypothesized value
n <- 30          # sample size

z <- (xbar - p)/sqrt((p*p)/n)
z

[1] 2.19089

5. Draw conclusion
z = 2.19 is in the rejectino region , __therfore we can reject the null hypthesis at 0.05 significance level__

##### apprach 2

In [33]:
prop.test(21, 30, correct = FALSE, alternative = "greater")


	1-sample proportions test without continuity correction

data:  21 out of 30, null probability 0.5
X-squared = 4.8, df = 1, p-value = 0.01423
alternative hypothesis: true p is greater than 0.5
95 percent confidence interval:
 0.5506175 1.0000000
sample estimates:
  p 
0.7 


The value of the test statistic, 4.8, is outside the rejection regions set by the critical values. Running a confidence interval hypothesis testing, the value stated by H0 is outside the confidence interval, therefore we can reject H0 and accept H1.